pkg/agent/config/config.go

```diff
@@ -461,7 +461,7 @@ func get(envInfo *cmds.Agent, proxy proxy.Proxy) (*config.Node, error) {
 	}
 
 	if !nodeConfig.Docker && nodeConfig.ContainerRuntimeEndpoint == "" {
-		nodeConfig.AgentConfig.RuntimeSocket = nodeConfig.Containerd.Address
+		nodeConfig.AgentConfig.RuntimeSocket = "unix://" + nodeConfig.Containerd.Address
```

Comment:
if the user specifies a url scheme on the cli this is going to cause problems, no?

Reply:
No, this is only used if the ContainerRuntimeEndpoint is an empty string ie not specified by the user.